src/ConstructiveSolidGeometry/VolumePrimitives/Cone.jl

```diff
@@ -4,9 +4,9 @@ struct Cone{T,TR,TP,TZ} <: AbstractVolumePrimitive{T}
     z::TZ
     #if r is a Tuple, the first entry refers to the r-interval at the bottom, the second one to the r-interval at the top
     function Cone( ::Type{T},
-                   r::Union{T, <:AbstractInterval{T}, Tuple{T,T}, Tuple{<:AbstractInterval{T},<:AbstractInterval{T}}},
+                   r::Union{T, <:AbstractInterval{T}, Tuple{T,T}, Tuple{I,I}},
                    φ::Union{Nothing, <:AbstractInterval{T}},
-                   z::Union{T, <:AbstractInterval{T}}) where {T}
+                   z::Union{T, <:AbstractInterval{T}}) where {T, I<:AbstractInterval{T}}
```

Comment:
maybe
```julia
r::Union{T, I, Tuple{T,T}, Tuple{I,I}},
φ::Union{Nothing, I}
z::Union{T, I}) where {T, I<:AbstractInterval{T}}
```
?

Reply:
Are we sure that all intervals are of the same type `(:closed,:closed)`?
This might not allow using `(:closed,:open)` for `φ` for example (not that we are using this right now, but who knows)

Reply:
ah, year.. You are right.